doc/architecture/decisions/0006-make-lists-a-thing.md

```diff
@@ -0,0 +1,106 @@
+# 6. Make named lists a first class construct
+
+## Context
+
+The current approach to collecting a list of things e.g. people that live in a household, and asking questions for each thing in that list, is more complex than it needs to be.
+
+### Collecting a list of things
+
+This is currently achieved using repeat until logic with a confirmation question to get out of the repeat - something like "Is this everyone that lives at ...?". While the happy path journey is acceptable, it soon falls apart when a user attempts to do something else, for example editing a list item or removing someone. This also means that every time we evaluate the routing path, we need to follow the list item entry and confirmation combo for every item in the list. Attempting to fit creation of a list of things into the standard question per block model is wrong, as the add/edit/remove journeys are not relevant outside of the list creation.
+
+### Asking questions for each thing in the list
+
+Early on in the development of runner the concept of repeating a group of questions based on answer count was implemented. This got us so far, but relies on isolating one answer to repeat on.
+Sometimes the list comes from more than one part of the questionnaire, for example there could be one set of repeats for collecting householders and another for collecting people who may be living away.
+
+## Decision
+
+- Define list creation in the schema
+- Store lists and answers within the context of a list
+- Remove all unnecessary repeat functionality
+
+### Define list creation in the schema
+
+A new block type will co-ordinate the collection of a list of things:
+
+```json
+{
+    "id": "block-id",
+    "type": "ListCollector",
+    "populates_list": "people",
+    "question": {},
+    "add_question": {},
+    "edit_question": {},
+    "remove_question": {}
+}
+```
+
+- The routing path is kept simple as we can ignore the journeys taken adding/editing/removing when evaluating the path.
+- Each question is a standard question type.
+- Each question can have variants. This is important as the add question can change depending on whether the list is empty or not.
```

Comment:
How would the above example support variants? 
https://github.com/ONSdigital/eq-schema-validator/pull/113/files#diff-87a1851b9e8803d68cb8195b642ee9a6R29
If it was like the linked example would we need `add_variants`?

Reply:
Yes, something like `add_variants` would fit with how question variants work.

Reply:
Given that we could have question variants as well here, we could look at changing that from `variants` to `question_variants`, so that if it was used in combination with add question variants it would be clear, something like:

```json
{
  "question_variants": {},
  "add_question_variants": [],
  "edit_question": {}
}
```

Another option is we could consider making the top level a block rather than a question, something like:

```json
{
  "question": {},
  "add_block": {
     "id": "add-something",
	 "question_variants": []
  },
  "edit_block": {
     "id": "edit-something",
	 "question": {}
  }
}
```

Reply:
Thinking about this a bit more, I also don't think we can enforce the mutual exclusivity of `add_question` / `add_question_variants` etc at the block leve in JSON schema.

Reply:
I think I prefer the block approach if only because it keeps everything more consistent and avoids another set of keys. 

I made a comment about the `question_variants` change on the validator PR, but I think it's probably worth doing.